utils/build/docker/golang/parametric/datadog.go

```diff
@@ -101,6 +106,56 @@ func (s *apmClientServer) SpanSetError(ctx context.Context, args *SpanSetErrorAr
 	return &SpanSetErrorReturn{}, nil
 }
 
+type CustomLogger struct {
+	*logrus.Logger
+}
+
+func (l *CustomLogger) Log(logMessage string) {
+	start := strings.Index(logMessage, "{")
+	end := strings.LastIndex(logMessage, "}")
+
+	if start == -1 || end == -1 {
+		fmt.Println("JSON not found in log message")
+		return
+	}
+	jsonStr := logMessage[start : end+1]
+	var config map[string]interface{}
+	if err := json.Unmarshal([]byte(jsonStr), &config); err != nil {
+		fmt.Printf("Error unmarshaling JSON: %v\n", err)
+		return
+	}
+	stringConfig := make(map[string]string)
+	for key, value := range config {
+		stringConfig[key] = fmt.Sprintf("%v", value)
+	}
+	globalConfig = stringConfig
+}
+
+func (s *apmClientServer) GetTraceConfig(ctx context.Context, args *GetTraceConfigArgs) (*GetTraceConfigReturn, error) {
+	var log = &CustomLogger{logrus.New()}
+	tracer.Start(tracer.WithLogger(log))
+
+	tracerEnabled := "true"
+	// if globalConfig is empty, then there were no startup logs generated and thus it means the tracer was disabled
+	if len(globalConfig) == 0 {
+		tracerEnabled = "false"
+	}
+	config := make(map[string]string)
+	config["dd_service"] = globalConfig["service"]
+	config["dd_log_level"] = "null" // golang doesn't support DD_LOG_LEVEL, only thing it supports is passing in DEBUG or DD_TRACE_DEBUG to set debug to true
+	config["dd_trace_sample_rate"] = globalConfig["sample_rate"]
+	config["dd_trace_enabled"] = tracerEnabled
+	config["dd_runtime_metrics_enabled"] = globalConfig["runtime_metrics_enabled"]
+	config["dd_tags"] = globalConfig["tags"]
+	config["dd_trace_propagation_style"] = globalConfig["propagation_style_inject"]
+	config["dd_trace_debug"] = globalConfig["debug"]
+	config["dd_trace_otel_enabled"] = "null"         // golang doesn't support DD_TRACE_OTEL_ENABLED
+	config["dd_trace_sample_ignore_parent"] = "null" // golang doesn't support DD_TRACE_SAMPLE_IGNORE_PARENT
```

Comment:
We should avoid setting the configuration to null. If the configuration can not be set then we should avoid setting in the config dictionary. Also null isn't idiomatic to golang. Nil would be more appropriate here 

```suggestion
	// config["dd_trace_sample_ignore_parent"] = nil // golang doesn't support DD_TRACE_SAMPLE_IGNORE_PARENT
```